examples/tracing/undump.py

```diff
@@ -0,0 +1,200 @@
+from bcc.utils import printb
+import argparse
+from socket import inet_ntop, ntohs, AF_INET, AF_INET6
+from struct import pack
+from time import sleep
+from datetime import datetime
+import sys
+
+# arguments
+examples = """examples:
+    ./undump           # trace/dump all UNIX packets
+    ./undump -t        # include timestamps
+    ./undump -p 181    # only trace/dump PID 181
+"""
+parser = argparse.ArgumentParser(
+    description="Dump UNIX socket packets",
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog=examples)
+    
+parser.add_argument("-t", "--timestamp", 
+        action="store_true", help="include timestamp on output")
+parser.add_argument("-p", "--pid",
+        help="trace this PID only")
+args = parser.parse_args()
+
+# define BPF program
+bpf_text = """
+#include <uapi/linux/ptrace.h>
+#include <net/sock.h>
+#include <bcc/proto.h>
+#include <linux/aio.h>
+#include <linux/socket.h>
+#include <linux/net.h>
+#include <linux/fs.h>
+#include <linux/mount.h>
+#include <linux/module.h>
+#include <net/sock.h>
+#include <net/af_unix.h>
+
+// separate data structs for ipv4 and ipv6
+struct stream_data_t {
+    u64 ts_us;
+    u32 pid;
+    u32 uid;
+    u32 sock_state;
+    u32 sock_type;  //type of socket[STREAM|DRGMA]
+    u64 sock_flags;
+    char task[TASK_COMM_LEN];
+    char *unix_sock_path;
+    int msg_namelen;
+};
+BPF_PERF_OUTPUT(stream_recvmsg_events);
+
+#define MAX_PKT 512
+struct recv_data_t {
+    u32 recv_len;
+    u8  pkt[MAX_PKT];
+};
+
+// single element per-cpu array to hold the current event off the stack
+BPF_PERCPU_ARRAY(unix_data, struct recv_data_t,1);
+
+BPF_PERF_OUTPUT(unix_recv_events);
+
+//static int unix_stream_recvmsg(struct socket *sock, struct msghdr *msg,
+//			       size_t size, int flags)
+int trace_stream_entry(struct pt_regs *ctx)
+{
+    int ret = PT_REGS_RC(ctx);
+    
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    u32 pid = pid_tgid >> 32;
+    u32 tid = pid_tgid;
+
+    FILTER_PID
+
+    struct stream_data_t data4 = {.pid = pid,};
+    data4.uid = bpf_get_current_uid_gid();
+    data4.ts_us = bpf_ktime_get_ns() / 1000;
+
+    struct socket *sock = (struct socket *)PT_REGS_PARM1(ctx);   
+    struct msghdr *msg = (struct msghdr *)PT_REGS_PARM2(ctx);    
+
+    data4.sock_state = sock->state;
+    data4.sock_type = sock->type;
+    data4.sock_flags = sock->flags;
+
+    data4.msg_namelen = msg->msg_namelen;
+    
+    bpf_get_current_comm(&data4.task, sizeof(data4.task));
+    
+    struct unix_sock *unsock = (struct unix_sock *)sock->sk;
+    data4.unix_sock_path = (char *)unsock->path.dentry->d_name.name;
+    
+    stream_recvmsg_events.perf_submit(ctx, &data4, sizeof(data4));
+
+    return 0;
+};
+
+int trace_unix_stream_read_actor(struct pt_regs *ctx)
+{
+    u32 zero = 0;
+    int ret = PT_REGS_RC(ctx);
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    u32 pid = pid_tgid >> 32;
+    u32 tid = pid_tgid;
+
+    FILTER_PID
```

Comment:
You have `FILTER_PID` here which is supposed to run under process context, so `bpf_get_current_comm` should also work, right ?

Reply:
You are right, i'll modify it

Reply:
I think this is good to go, pending this comment being addressed. @chenhengqi do you think it's fine to merge w/o another iteration?

Reply:
Yeah, since this tool is served as an example not a BCC tool, I think it is good to go.